modules/exploits/linux/local/polkit_priv_esc.rb

```diff
@@ -0,0 +1,254 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'unix_crypt'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Kernel
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Local::Linux
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Polkit Authentication Bypass',
+        'Description' => %q{
+          A vulnerability exists within the polkit system service that can be leveraged by a local, unprivileged
+          attacker to perform privileged operations. In order to leverage the vulnerability, the attacker invokes a
+          method over D-Bus and kills the client process. This will occasionally cause the operation to complete without
+          being subjected to all of the necessary authentication.
+          The exploit module leverages this to add a new user with a sudo access and a known password. The new account
+          is then leveraged to execute a payload with root privileges.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Kevin Backhouse',                   # vulnerability discovery and analysis
+            'Spencer McIntyre',                  # metasploit module
+            'jheysel-r7'                         # metasploit module
+          ],
+        'SessionTypes' => ['shell', 'meterpreter'],
+        'Platform' => ['unix', 'linux'],
+        'References' => [
+          ['URL', 'https://github.blog/2021-06-10-privilege-escalation-polkit-root-on-linux-with-bug/'],
+          ['CVE', '2021-3560'],
+        ],
+        'Targets' =>
+          [
+            [ 'Automatic', {} ],
+          ],
+        'DefaultTarget' => 0,
+        # DefaultOptions' => { 'PrependSetgid' => true, 'PrependSetuid' => true, 'WfsDelay' => 10 },
+        'DisclosureDate' => '2021-06-03',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [ARTIFACTS_ON_DISK, CONFIG_CHANGES, IOC_IN_LOGS],
```

Comment:
Presumably this may also spam pkexec agent authentication dialogs.

```suggestion
          'SideEffects' => [ARTIFACTS_ON_DISK, CONFIG_CHANGES, IOC_IN_LOGS, SCREEN_EFFECTS],
```